src/hw1/task2.js

```diff
@@ -0,0 +1,32 @@
+import * as fs from 'node:fs';
+import csvtojson from 'csvtojson';
+import  { pipeline } from 'node:stream';
+
+const readStream = fs.createReadStream('./assets/nodejs-hw1-ex1.csv');
+const writeStream = fs.createWriteStream('./gen/output1.txt');
+const writeStream2 = fs.createWriteStream('./gen/output2.txt');
+const converter = csvtojson({ delimiter: ';' });
+
+pipeline(
```

Comment:
using `readline` the implementation might look like this:
```
const writeStream = fs.createWriteStream('./output.txt');

const rl = readline.createInterface({
    input: fs.createReadStream('./file.csv'),
    output: writeStream,
    terminal: false
});

rl.on('line', async (line) =>  {
    const jsonLine = await parseLine(line);
    writeStream.write(JSON.stringify(jsonLine) + EOL);
});

rl.on('error', (err) => {
    console.error(err.message);
});
```
I've not found in nodejs docs that `readline` can be used as source stream, however the following code seem to work:

```
const rl = readline.createInterface({
    input: fs.createReadStream('./file.csv'),
    terminal: false
});

const transformStream = new Transform({
    async transform(chunk, encoding, callback) {
        const csvLine = Buffer.from(chunk).toString('utf8');

        const jsonLine = await parseLine(csvLine);

        callback(null, JSON.stringify(jsonLine) + EOL);
    }
})


pipeline(
    rl,
    transformStream,
    fs.createWriteStream('./output_pipeline.txt'),
    (err) => {
        if (err) {
            console.error(err.message)
        } else {
            console.log('Done!')
        }
    }
)
```